DOC-4544 added PHP vector query example (#1431)
redis/docs: content/develop/clients/php/vecsearch.md

[`all-mpnet-base-v2`](https://huggingface.co/sentence-transformers/all-mpnet-base-v2)
to generate the vector embeddings to store and index with Redis Query Engine.

## Initialize

You can use the [TransformersPHP](https://transformers.codewithkyrian.com/)
library to create the vector embeddings. Install the library with the following
command:

```bash
composer require codewithkyrian/transformers
```

## Import dependencies

Import the following classes and function in your source file:

```php
<?php

require 'vendor/autoload.php';

// TransformersPHP
use function Codewithkyrian\Transformers\Pipelines\pipeline;

// Redis client and query engine classes.
use Predis\Client;
use Predis\Command\Argument\Search\CreateArguments;
use Predis\Command\Argument\Search\SearchArguments;
use Predis\Command\Argument\Search\SchemaFields\TextField;
use Predis\Command\Argument\Search\SchemaFields\TagField;
use Predis\Command\Argument\Search\SchemaFields\VectorField;
```

## Create a tokenizer instance

The code below shows how to use the
[`all-mpnet-base-v2`](https://huggingface.co/sentence-transformers/all-mpnet-base-v2)
tokenizer to generate the embeddings. The vectors that represent the

> embeddings have 384 components, regardless of the length of the input

Reviewer: Previous versions of this document used dimensions; I think that's better than components.

```suggestion
embeddings have 384 dimensions, regardless of the length of the input
```

text. Here, the `pipeline()` call creates the `$extractor` function that
generates embeddings from text:

```php

> $extractor = pipeline('embeddings', 'Xenova/all-MiniLM-L6-v2');

Reviewer: Shouldn't this be Xenova/all-mpnet-base-v2?

```suggestion
$extractor = pipeline('embeddings', 'Xenova/all-mpnet-base-v2');
```

Author: @dwdougherty Actually, this is the only instance that's right! (It works in the tested PHP code.) I accidentally used the wrong model name in the text by taking it from another example that uses a different model :-( Fixed now, though - thanks for spotting the inconsistency.

$client->hmset("doc:2",[
    "content" => $content,
    "genre" => "weather",
    "embedding" => pack('g*', ...$emb[0])
]);
```

## Run a query

After you have created the index and added the data, you are ready to run a query.
To do this, you must create another embedding vector from your chosen query
text. Redis calculates the vector distance between the query vector and each
embedding vector in the index as it runs the query. You can request the results to be
sorted to rank them in order of ascending distance.

The code below creates the query embedding using the `$extractor()` function, as with
the indexing, and passes it as a parameter when the query executes (see
[Vector search]({{< relref "/develop/interact/search-and-query/query/vector-search" >}})
for more information about using query parameters with embeddings).
The query is a
[K nearest neighbors (KNN)]({{< relref "/develop/interact/search-and-query/advanced-concepts/vectors#knn-vector-search" >}})
search that sorts the results in order of vector distance from the query vector.

The results are returned as an array with the number of results in the
first element. The remaining elements are alternating pairs with the

> key of the returned document (eg, `doc:0`) first, followed by an array containing

Reviewer: SG

```suggestion
key of the returned document (for example, `doc:0`) first, followed by an array containing
```
